rfcs/0007-additional-type-representations.md

```diff
@@ -0,0 +1,48 @@
+# Additional Type Representations
+
+## Purpose
+
+The current list of scalar type representations is inadequate:
+
+- JSON integer and number types are not what we want for most applications, since they are represented by an unusual 53-bit number type. Since scalars can be both inputs and outputs, and are therefore invariant, we cannot safely convert to and from larger and smaller integer and floating point types.
+- We want to be able to generate some standard scalar types in Hasura for types which are used across different data sources, e.g. UUIDs. This needs support in NDC.
+
+## Proposal
+
+Add the following type representations to `ndc_models::TypeRepresentation`:
+
+|Name|Description|JSON representation|
+|-|-|-|
+| Int8 | A 8-bit signed integer with a minimum value of -2^7 and a maximum value of 2^7 - 1 | Number |
+| Int16 | A 16-bit signed integer with a minimum value of -2^15 and a maximum value of 2^7 - 1 |Number |
+| Int32 | A 32-bit signed integer with a minimum value of -2^31 and a maximum value of 2^7 - 1 |Number |
+| Int64 | A 64-bit signed integer with a minimum value of -2^63 and a maximum value of 2^7 - 1 |String |
+| Float32 | An IEEE-754 single-precision floating-point number | Number |
+| Float64 | An IEEE-754 double-precision floating-point number | String |
+| Decimal | | String |
+| Char | A unicode scalar value |
+| Uuid | | String |
```

Comment:
IMO,  I prefer `UUID` instead
`Int` already represents `Int32`.  We don't need to define another type.  `Number` is also duplicated with `Float64`. Should we deprecate it?
FYI this is also a good reference https://the-guild.dev/graphql/scalars/docs/scalars

Reply:
`Int` is a JSON 53-bit integer, and since types can be used as inputs and outputs, the concern was that these are semantically imprecise.

A connector author could choose to use `Int` for a 32-bit integer, but would have to throw an error if that type were reused as an input (e.g. in a predicate) and an out-of-range value were provided.

I actually think that's a reasonable trade-off for the simplicity, but we can discuss it here.

Reply:
After a discussion, I have updated the spec to remove `Int` and `Float`.